girder/api/rest.py

```diff
@@ -376,13 +376,16 @@ def _setCommonCORSHeaders(isOptions=False):
     origin = '%s://%s' % (originparse.scheme, originparse.netloc)
     cherrypy.response.headers['Access-Control-Allow-Origin'] = origin
     cherrypy.response.headers['Vary'] = 'Origin'
-    # Some requests do not require further checking
-    if (cherrypy.request.method in ('GET', 'HEAD') or (
-            cherrypy.request.method == 'POST' and cherrypy.request.headers.get(
-            'Content-Type', '').split(';', 1)[0].strip() in (
-                'application/x-www-form-urlencoded',
-                'multipart/form-data', 'text/plain'))):
-        return
+    # Some requests are "simple", and don't require the browser to have sent an
+    # OPTIONS request before them.  If we want to be stricter, we would log
+    # what OPTIONS requests have occurered, and only respond to non-simple
+    # requests that had been preflighted.
```

Comment:
We *can* rely on the browser, since a non-compliant browser could always just not send an `ORIGIN` header.  To be the most pedantic, though, we would enforce getting an `OPTIONS` preflight before another method.  I don't think it is really necessary as only non-complaint browsers would be affected.